src/rpc/rpc.cc

```diff
@@ -1170,16 +1170,27 @@ future<> server::connection::send_unknown_verb_reply(std::optional<rpc_clock_typ
       });
   }
 
-  future<> server::stop() {
+  future<> server::shutdown() {
+      if (_shutdown) {
+          return make_ready_future<>();
+      }
+
       _ss.abort_accept();
       _resources_available.broken();
       if (_options.streaming_domain) {
           _servers.erase(*_options.streaming_domain);
       }
-      return when_all(_ss_stopped.get_future(),
+      return
           parallel_for_each(_conns | boost::adaptors::map_values, [] (shared_ptr<connection> conn) {
               return conn->stop();
-          }),
+          }).finally([this] {
+              _shutdown = true;
+          });
+  }
+
+  future<> server::stop() {
+      return when_all(_ss_stopped.get_future(),
+          shutdown(),
```

Comment:
Can we not have two copies of shutdown() running? Or is the user required to wait for shutdown first (sensible)?